ocpmodels/models/dimenet_plus_plus.py

```diff
@@ -0,0 +1,449 @@
+        num_radial,
+        cutoff=5.0,
+        envelope_exponent=5,
+        num_before_skip=1,
+        num_after_skip=2,
+        num_output_layers=3,
+        act=swish,
+    ):
+        super(DimeNetPlusPlus, self).__init__()
+
+        self.cutoff = cutoff
+
+        if sym is None:
+            raise ImportError("Package `sympy` could not be found.")
+
+        self.num_blocks = num_blocks
+
+        self.rbf = BesselBasisLayer(num_radial, cutoff, envelope_exponent)
+        self.sbf = SphericalBasisLayer(
+            num_spherical, num_radial, cutoff, envelope_exponent
+        )
+
+        self.emb = EmbeddingBlock(num_radial, hidden_channels, act)
+
+        self.output_blocks = torch.nn.ModuleList(
+            [
+                OutputPPBlock(
+                    num_radial,
+                    hidden_channels,
+                    out_emb_channels,
+                    out_channels,
+                    num_output_layers,
+                    act,
+                )
+                for _ in range(num_blocks + 1)
+            ]
+        )
+
+        self.interaction_blocks = torch.nn.ModuleList(
+            [
+                InteractionPPBlock(
+                    hidden_channels,
+                    int_emb_size,
+                    basis_emb_size,
+                    num_spherical,
+                    num_radial,
+                    num_before_skip,
+                    num_after_skip,
+                    act,
+                )
+                for _ in range(num_blocks)
+            ]
+        )
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self.rbf.reset_parameters()
+        self.emb.reset_parameters()
+        for out in self.output_blocks:
+            out.reset_parameters()
+        for interaction in self.interaction_blocks:
+            interaction.reset_parameters()
+
+    def triplets(self, edge_index, num_nodes):
+        row, col = edge_index  # j->i
+
+        value = torch.arange(row.size(0), device=row.device)
+        adj_t = SparseTensor(
+            row=col, col=row, value=value, sparse_sizes=(num_nodes, num_nodes)
+        )
+        adj_t_row = adj_t[row]
+        num_triplets = adj_t_row.set_value(None).sum(dim=1).to(torch.long)
+
+        # Node indices (k->j->i) for triplets.
+        idx_i = col.repeat_interleave(num_triplets)
+        idx_j = row.repeat_interleave(num_triplets)
+        idx_k = adj_t_row.storage.col()
+        mask = idx_i != idx_k  # Remove i == k triplets.
+        idx_i, idx_j, idx_k = idx_i[mask], idx_j[mask], idx_k[mask]
+
+        # Edge indices (k-j, j->i) for triplets.
+        idx_kj = adj_t_row.storage.value()[mask]
+        idx_ji = adj_t_row.storage.row()[mask]
+
+        return col, row, idx_i, idx_j, idx_k, idx_kj, idx_ji
+
+    def forward(self, z, pos, batch=None):
+        """"""
+        raise NotImplementedError
+
+
+@registry.register_model("dimenetplusplus")
+class DimeNetPlusPlusWrap(DimeNetPlusPlus):
```

Comment:
Should we also support `otf_graph` to be consistent with other models?

Reply:
Yes, good point! I'll add that